pkg/monitor/intervalcreation/state_tracker.go

```diff
@@ -0,0 +1,147 @@
+package intervalcreation
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/openshift/origin/pkg/monitor/monitorapi"
+	"k8s.io/apimachinery/pkg/util/sets"
+)
```

Comment:
Unit tests might be good here, I know it's getting tested at a higher level but are we certain it's enough coverage?